lib/services/project-files-manager.ts

```diff
@@ -0,0 +1,56 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import util = require("util");
+
+export class ProjectFilesManager implements IProjectFilesManager {
+	constructor(private $fs: IFileSystem,
+		private $platformsData: IPlatformsData) { }
+	
+	public processPlatformSpecificFiles(directoryPath: string, platform: string, excludedDirs?: string[]) {
+		return (() => {
+			var contents = this.$fs.readDirectory(directoryPath).wait();
+			var files: string[] = [];
+			
+			_.each(contents, fileName => {
+				let filePath = path.join(directoryPath, fileName);
+				let fsStat = this.$fs.getFsStats(filePath).wait();
+				if(fsStat.isDirectory() && (!excludedDirs || (excludedDirs && !_.contains(excludedDirs, fileName)))) {
+					this.processPlatformSpecificFilesCore(platform, this.$fs.enumerateFilesInDirectorySync(filePath)).wait();
+				} else if(fsStat.isFile()) {
+					files.push(filePath);
+				}
+			});
+			this.processPlatformSpecificFilesCore(platform, files).wait();
```

Comment:
Maybe we can improve the performance here:
```TypeScript
let futures: IFuture[];
_.each(contents, fileName => {
	let filePath = path.join(directoryPath, fileName);
	let fsStat = this.$fs.getFsStats(filePath).wait();
	if(fsStat.isDirectory() && !_.contains(excludedDirs, fileName)) {
		future.push(this.processPlatformSpecificFilesCore(platform, this.$fs.enumerateFilesInDirectorySync(filePath)));
	} else if (fsStat.isFile()) {
		files.push(filePath);
});
futures.push(this.processPlatformSpecificFilesCore(platform, files));
Future.wait(futures);
```
I've not tested it and this is **not** a merge stopper.